elfo-core/src/init.rs

```diff
@@ -29,48 +29,50 @@ use crate::{
 
 type Result<T, E = StartError> = std::result::Result<T, E>;
 
-async fn start_entrypoints(ctx: &Context, topology: &Topology) -> Result<()> {
+async fn start_entrypoints(ctx: &Context, topology: &Topology, is_check_only: bool) -> Result<()> {
     let futures = topology
         .actor_groups()
         .filter(|group| group.is_entrypoint)
-        .map(|group| {
-            let config = Default::default();
-            ctx.request_to(group.addr, UpdateConfig { config })
-                .resolve()
-                .or_else(|err| async move {
-                    match err {
-                        RequestError::Ignored => Ok(Ok(())),
-                        _ => Err(StartError::Other(
+        .map(|group| async move {
+            fn convert_response<R>(
+                group: String,
+                response: Result<Result<(), impl Into<StartErrorReason>>, RequestError<R>>,
+            ) -> Result<()> {
+                match response {
+                    Ok(Ok(())) => Ok(()),
+                    Ok(Err(e)) => Err(StartError {
+                        group,
+                        reason: e.into(),
+                    }),
+                    Err(RequestError::Ignored) => Ok(()),
+                    Err(RequestError::Closed(..)) => Err(StartError {
+                        group,
+                        reason: StartErrorReason::Other(
                             "initial messages cannot be delivered".into(),
-                        )),
-                    }
-                })
-        });
-
-    let error_count = futures::future::try_join_all(futures)
-        .await?
-        .into_iter()
-        .filter_map(Result::err)
-        .count();
-
-    if error_count == 0 {
-        Ok(())
-    } else {
-        Err(StartError::InvalidConfig)
-    }
-}
+                        ),
+                    }),
+                }
+            }
 
-async fn wait_entrypoints(ctx: &Context, topology: &Topology) -> Result<()> {
```

Comment:
This is a proposal to remove `wait_entrypoints` step. Entrypoint is free to exit as soon as it finished doing its thing, so there can be a situation where `UpdateConfig` + `StartEntrypoint` are successful, but the `Ping` afterwards fails.